parser.go

```diff
@@ -405,18 +405,26 @@ func (p *parser) parseAttributeSelector() (attrSelector, error) {
 		return attrSelector{}, err
 	}
 
+	// check if the attribute contains an ignore case flag
+	ignoreCase := false
+	if p.s[p.i] == ' ' && p.s[p.i+1] == 'i' {
```

Comment:
This seems a little too simple. What if there are 2 spaces, or a tab? Or `in` instead of `i`?

Reply:
@andybalholm I agree that the parser is a little light. I read the [Mozilla MDN Page]() before implementing the parser. They say:

> [attr operator value i]
Adding an i (or I) before the closing bracket causes the value to be compared case-insensitively (for characters within the ASCII range).

So I only checked for a space and a `i` before the closing bracket. If you changed `i` to `in`, it raises an error (which I think it's normal):

>     selector_test.go:805: error compiling "[title*=\"BaRu\" in]": expected ']', found 'n' instead

Multiple spaces before the `I` also raises an error but not a clear one indeed:

>     selector_test.go:805: error compiling "[title$=\"BAR\"  i]": expected ']', found 'i' instead

Maybe I need to fix that. I think I should check more safely for a `i` before the closing bracket, preceded by a space. Should I take care of uppercase too for `i`?

Reply:
@andybalholm just added a code to handle multi spaces and uppercase "I" https://github.com/andybalholm/cascadia/pull/51/commits/b906f18252fef489b343c6fe1b6b6f45ae85f2bc